test/gtest/ucp/ucp_test.cc

```diff
@@ -213,10 +213,11 @@ void ucp_test::wait(void *req, int worker_index)
     }
 
     ucs_status_t status;
+    ucs_time_t deadline = ucs::get_deadline();
     do {
         progress(worker_index);
         status = ucp_request_check_status(req);
-    } while (status == UCS_INPROGRESS);
+    } while ((status == UCS_INPROGRESS) && (ucs_get_time() < deadline));
```

Comment:
to detect a hang

Reply:
but seems like this function (also the one below) is not called on a request which should be canceled? is it only to avoid gtest hang? because we already have a watchdog thread?

Reply:
> is it only to avoid gtest hang?

yes, we have deadline almost everywhere in request waiting loops, do you suggest don't use deadline and switch to watchdog for these cases as well? Does not deadline approach provide better bug localization in case of failure (hang)?

Reply:
ok, i guess it's fine since there is status check after do..while loop in both cases